test/unittest/dynamic_types/DynamicTypesTests.cpp

```diff
@@ -3698,893 +3705,703 @@ TEST_F(DynamicTypesTests, DynamicType_union_with_unions_unit_tests)
     ASSERT_TRUE(DynamicDataFactory::GetInstance()->IsEmpty());
 }
 
-TEST_F(DynamicTypesTests, DynamicType_XML_BoolStruct_test)
+TEST_F(DynamicTypesTests, DynamicType_XML_EnumStruct_test)
 {
     using namespace xmlparser;
     using namespace types;
-    const char* config_file = "types.xml";
-    tinyxml2::XMLDocument doc;
 
-    XMLP_ret ret = XMLProfileManager::loadXMLFile(config_file);
+    XMLP_ret ret = XMLProfileManager::loadXMLFile(DynamicTypesTests::config_file());
     ASSERT_EQ(ret, XMLP_ret::XML_OK);
     {
-        DynamicPubSubType *pbType = XMLProfileManager::CreateDynamicPubSubType("BoolStruct");
-        DynamicData* data = DynamicDataFactory::GetInstance()->CreateData(pbType->GetDynamicType());
-
-        // SERIALIZATION TEST
-        BoolStruct wbool;
-        BoolStructPubSubType wboolpb;
+        DynamicPubSubType *pbType = XMLProfileManager::CreateDynamicPubSubType("EnumStruct");
+        
+        DynamicTypeBuilderFactory* m_factory = DynamicTypeBuilderFactory::GetInstance();
 
-        uint32_t payloadSize = static_cast<uint32_t>(pbType->getSerializedSizeProvider(data)());
-        SerializedPayload_t payload(payloadSize);
-        SerializedPayload_t dynamic_payload(payloadSize);
-        ASSERT_TRUE(pbType->serialize(data, &dynamic_payload));
-        ASSERT_TRUE(wboolpb.deserialize(&dynamic_payload, &wbool));
+        // Enum
+        DynamicTypeBuilder_ptr enum_builder = m_factory->CreateEnumBuilder();
+        enum_builder->AddEmptyMember(0, "A");
+        enum_builder->AddEmptyMember(1, "B");
+        enum_builder->AddEmptyMember(2, "C");
+        enum_builder->SetName("MyEnum");
 
-        uint32_t static_payloadSize = static_cast<uint32_t>(wboolpb.getSerializedSizeProvider(&wbool)());
-        SerializedPayload_t static_payload(static_payloadSize);
-        ASSERT_TRUE(wboolpb.serialize(&wbool, &static_payload));
-        ASSERT_TRUE(static_payload.length == static_payloadSize);
+        // Struct EnumStruct
+        DynamicTypeBuilder_ptr es_builder = m_factory->CreateStructBuilder();
+        es_builder->AddMember(0, "my_enum", enum_builder.get());
+        es_builder->SetName("EnumStruct");
 
-        ASSERT_TRUE(DynamicDataFactory::GetInstance()->DeleteData(data) == ResponseCode::RETCODE_OK);
```

Comment:
Why we remove all testing of create data and serialize/deserialize?

Reply:
We decided to modify the nature of these tests because "DynamicComplexTypesTests" does already that checking more deeply and focus "DynamicTypesTests" to XML parsing.